buffer/src/main/java/io/netty/buffer/ByteBufAllocatorMetricProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+public interface ByteBufAllocatorMetricProvider {
```

Comment:
@Scottmitch I think it is more natural to only have the methods on the metric itself or on the allocator. As we said we wanted to expose a metric pojo we should be consistent and make use of it and not have these also on other objects IMHO.
 
I am not strong on the type thingy and can add it if you like... Just not sure its really needed as we can just override it as I did.